sidechain/consensus/common/src/error.rs

```diff
@@ -0,0 +1,64 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2017-2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Error types in sidechain Consensus
+
+use its_primitives::types::block::BlockHash;
+use std::{
+	boxed::Box,
+	error,
+	string::{String, ToString},
+	vec::Vec,
+};
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+pub use thiserror_sgx as thiserror;
+
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
```

Comment:
what does this annotation do?

Reply:
This is an interesting copy-paste addition. Internally, it does nothing. But when this enum is used in an external crate. The compiler will enforce to always have an `_ => do someting` match arm.

This should prevent some breaking changes in third party crates.